frontend/src/components/data-table/renderers.tsx

```diff
@@ -213,6 +259,81 @@ export const DataTableBody = <TData,>({
       )}
     </TableBody>
   );
+
+  return (
+    <ContextMenu onOpenChange={handleContextMenuChange}>
```

Comment:
maybe we can hide most of this behind `<DataTableContextMenu tableRef={tableRef}>{tableBody}...`